app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/CharacterServicesImpl.kt

```diff
@@ -22,4 +20,19 @@ class CharacterServicesImpl(private val api: MarvelResquestGenerator = MarvelRes
             }
         }
     }
+
+    fun getCaracters(id: Int): Observable<Character> {
+        return Observable.create { subscriber ->
+            val callResponse = api.createService(MarvelApi::class.java).getCharacterById(id)
+            val response = callResponse.execute()
+
+            if (response.isSuccessful) {
+                subscriber.onNext(mapper.transform(response.body()!!.data!!.characters.get(0)))
```

Comment:
no use "!!" you can use instead "?." or "?:"

Reply:
Done, Thanks